api/libproject/devstart.rb

```diff
@@ -1100,66 +1100,53 @@ def with_cloud_proxy_and_db_env(cmd_name, args)
-    return
-  end
+def deploy(cmd_name, args)
+  op = WbOptionsParser.new(cmd_name, args)
+  op.add_option(
+    "--version [version]",
+    lambda {|opts, v| opts.version = v},
+    "Version to deploy (e.g. your-username-test)"
+  )
+  op.add_validator lambda {|opts| raise ArgumentError unless opts.version}
+  op.add_option(
+    "--promote",
+    lambda {|opts, v| opts.promote = true},
+    "Promote this version to immediately begin serving API traffic"
+  )
+  op.add_option(
+    "--no-promote",
+    lambda {|opts, v| opts.promote = false},
+    "Deploy, but do not yet serve traffic from this version - DB migrations are still applied"
+  )
+  op.add_validator lambda {|opts| raise ArgumentError if opts.promote.nil?}
+
+  gcc = GcloudContextV2.new(op)
+  op.parse.validate
+  gcc.validate
 
   unless Workbench::in_docker?
     exec *(%W{docker run --rm -v #{File.expand_path("..")}:/w -w /w/api
       allofustest/workbench:buildimage-0.0.9
       ./project.rb #{cmd_name}} + args)
   end
 
-  common = Common.new
-  unless File.exist? "circle-sa-key.json"
-    common.error "Missing service account credentials file circle-sa-key.json."
-    exit 1
-  end
-
-  if is_master
-    common.status "Running database migrations..."
-    with_cloud_proxy_and_db_env(cmd_name, args) do |ctx|
-      migrate_database
-      load_config(ctx.project)
-
-      common.status "Pushing GCS artifacts..."
-      deploy_gcs_artifacts(cmd_name, %W{--project #{ctx.project}})
-    end
-  end
+  common.status "Running database migrations..."
+  with_cloud_proxy_and_db(gcc) do |ctx|
+    migrate_database
+    load_config(ctx.project)
 
-  promote = ""
-  version = ""
-  if is_master
-    # Note that --promote will generally be a no-op, as we expect
-    # circle-ci-test to always be serving 100% traffic. Pushing to an existing
-    # live version will immediately make those changes live. In the event that
-    # someone mistakenly pushes a different version manually, this --promote
-    # will restore us to the expected circle-ci-test version on the next commit.
-    promote = "--promote"
-    version = "circle-ci-test"
-  elsif ENV.has_key?("CIRCLE_TAG")
-    promote = "--no-promote"
-    version = ENV["CIRCLE_TAG"]
+    common.status "Pushing GCS artifacts..."
+    deploy_gcs_artifacts(cmd_name, %W{--project #{ctx.project}})
   end
 
-  deploy_api(cmd_name, args + %W{--quiet --version #{version} #{promote}})
-  deploy_public_api(cmd_name, args + %W{--quiet --version #{version} #{promote}})
+  deploy_api(cmd_name, %W{--quiet --version #{op.opts.version} #{promote}})
+  deploy_public_api(cmd_name, %W{--quiet --version #{op.opts.version} #{promote}})
 end
 
 Common.register_command({
-  :invocation => "circle-deploy",
-  :description => "Deploys the API server from within the Circle CI envronment.",
-  :fn => lambda { |*args| circle_deploy("circle-deploy", args) }
+  :invocation => "deploy",
+  :description => "Run DB migrations and deploy the API and public servers",
```

Comment:
Can you update the README to indicate how to use this for manual deploys?

Reply:
Done. Separately there is more work to do here on defining a process around the next stages in deployment.  Ideally we'd improve on our current story for RDR releases (using a local client), and we also should figure out what SA we think this is going to be (or if it's your user account).